homeassistant/components/dnsip/config_flow.py

```diff
@@ -178,15 +202,19 @@ async def async_step_init(
                     title=self.config_entry.title,
                     data={
                         CONF_RESOLVER: resolver,
+                        CONF_PORT: port,
                         CONF_RESOLVER_IPV6: resolver_ipv6,
+                        CONF_PORT_IPV6: port_ipv6,
                     },
                 )
 
         schema = self.add_suggested_values_to_schema(
             vol.Schema(
                 {
                     vol.Optional(CONF_RESOLVER): cv.string,
+                    vol.Optional(CONF_PORT): cv.port,
```

Comment:
Should we have a default here too so the user can't leave the field empty and thereby not get a value?

Reply:
I was following the lead of https://github.com/home-assistant/core/pull/114058 where defaults were removed.

Presumably because they were unnecessary duplicates as defaults are defined here: https://github.com/home-assistant/core/pull/113993/files#diff-f7e2ef265b774da4533e59e774aad087920f82fc97c3b9e7801f15537360bb5dR47-R51

Reply:
You're probably right so let's not have the default in the schema for the options flow. Just have a check so if left empty we set it to `53`.

Reply:
Isn't that achieved by: https://github.com/home-assistant/core/blob/fabfea5c057d77df97f31a8b691f75cec58a7227/homeassistant/components/dnsip/config_flow.py#L118-L119
As user omitted inputs are replaced by defaults?

Reply:
Tested at [c9e6f5b](https://github.com/home-assistant/core/pull/113993/commits/c9e6f5b1606630cf9338f8588b6ef305ee196ae9)

https://github.com/home-assistant/core/assets/11440490/d6e8585f-edd1-44d5-9427-ab91e70e6935